cmd/zed/zed.d/zed-functions.sh

```diff
@@ -438,3 +438,23 @@ zed_guid_to_pool()
 		$ZPOOL get -H -ovalue,name guid | awk '$1=='"$guid"' {print $2}'
 	fi
 }
+
+# zed_exit_if_ignoring_this_event
+#
+# Exit the script if we should ignore this event, as determined by
+# $ZED_SYSLOG_SUBCLASS_INCLUDE and $ZED_SYSLOG_SUBCLASS_EXCLUDE in zed.rc.
+# This function assumes you've imported the normal zed variables.
+zed_exit_if_ignoring_this_event
+{
+	if [ -n "${ZED_SYSLOG_SUBCLASS_INCLUDE}" ]; then
+	    case "${ZEVENT_SUBCLASS}" in
+	    ${ZED_SYSLOG_SUBCLASS_INCLUDE});;
```

Comment:
Unfortunately this doesn't seem to work with multiple wildcards, see discussion here https://github.com/zfsonlinux/zfs/pull/7007/files/7ace2363d44535fff6480632f69c6015d97ba588#diff-2c32bcdfcd8000007579664d4ec53bd1

```
root@linux:~# cat zed-test.sh 
#!/bin/sh

. /etc/zfs/zed.d/zed-functions.sh

echo ZED_SYSLOG_SUBCLASS_INCLUDE=$ZED_SYSLOG_SUBCLASS_INCLUDE
echo ZED_SYSLOG_SUBCLASS_EXCLUDE=$ZED_SYSLOG_SUBCLASS_EXCLUDE
echo ZEVENT_SUBCLASS=$ZEVENT_SUBCLASS

zed_exit_if_ignoring_this_event

echo "logged"

root@linux:~# ZEVENT_SUBCLASS=history_event ./zed-test.sh
ZED_SYSLOG_SUBCLASS_INCLUDE=
ZED_SYSLOG_SUBCLASS_EXCLUDE=
ZEVENT_SUBCLASS=history_event
logged
root@linux:~# ZED_SYSLOG_SUBCLASS_INCLUDE='history_*' ZEVENT_SUBCLASS=history_event ./zed-test.sh
ZED_SYSLOG_SUBCLASS_INCLUDE=history_*
ZED_SYSLOG_SUBCLASS_EXCLUDE=
ZEVENT_SUBCLASS=history_event
logged
root@linux:~# ZED_SYSLOG_SUBCLASS_INCLUDE='history_*|pool_*' ZEVENT_SUBCLASS=history_event ./zed-test.sh
ZED_SYSLOG_SUBCLASS_INCLUDE=history_*|pool_*
ZED_SYSLOG_SUBCLASS_EXCLUDE=
ZEVENT_SUBCLASS=history_event
root@linux:~# 
```